psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -0,0 +1,766 @@
+                diis_performed = self.diis()
+
+            if diis_performed:
+                status.append("DIIS")
+
+            core.timer_off("HF: DIIS")
+
+            if verbose > 4 and diis_performed:
+                core.print_out("  After DIIS:\n")
+                self.Fa().print_out()
+                self.Fb().print_out()
+
+            # frac, MOM invoked here from Wfn::HF::find_occupation
+            core.timer_on("HF: Form C")
+            self.form_C()
+            core.timer_off("HF: Form C")
+
+        if self.MOM_performed_:
+            status.append("MOM")
+
+        if self.frac_performed_:
+            status.append("FRAC")
+
+        core.timer_on("HF: Form D")
+        self.form_D()
+        core.timer_off("HF: Form D")
+
+        core.set_variable("SCF ITERATION ENERGY", SCFE)
+
+        # After we've built the new D, damp the update
+        if (damping_enabled and self.iteration_ > 1 and Drms > core.get_option('SCF', 'DAMPING_CONVERGENCE')):
+            damping_percentage = core.get_option('SCF', "DAMPING_PERCENTAGE")
+            self.damping_update(damping_percentage * 0.01)
+            status.append("DAMP={}%".format(round(damping_percentage)))
+
+        if verbose > 3:
+            self.Ca().print_out()
+            self.Cb().print_out()
+            self.Da().print_out()
+            self.Db().print_out()
+
+        # Print out the iteration
+        core.print_out("   @%s%s iter %3d: %20.14f   %12.5e   %-11.5e %s\n" %
+                       ("DF-" if is_dfjk else "", reference, self.iteration_, SCFE, Ediff, Drms, '/'.join(status)))
+
+        # if a an excited MOM is requested but not started, don't stop yet
+        if self.MOM_excited_ and not self.MOM_performed_:
+            continue
+
+        # if a fractional occupation is requested but not started, don't stop yet
+        if frac_enabled and not self.frac_performed_:
+            continue
+
+        # Call any postiteration callbacks
+
+        if _converged(Ediff, Drms, e_conv=e_conv, d_conv=d_conv):
+            break
+        if self.iteration_ >= core.get_option('SCF', 'MAXITER'):
+            raise ConvergenceError("""SCF iterations""", self.iteration_)
+
+
+def scf_finalize_energy(self):
+    """Performs stability analysis and calls back SCF with new guess
+    if needed, Returns the SCF energy. This function should be called
+    once orbitals are ready for energy/property computations, usually
+    after iterations() is called.
+
+    """
+
+    # post-scf vv10 correlation
+    if core.get_option('SCF', "DFT_VV10_POSTSCF"):
+        self.functional().set_lock(False)
+        self.functional().set_do_vv10(True)
+        self.functional().set_lock(True)
+        core.print_out("  ==> calculating VV10 correction on post-scf density <==\n\n")
+        SCFE = 0.0
+        self.form_V()
+        SCFE += self.compute_E()
+        self.set_energies("Total Energy", SCFE)
```

Comment:
Pretty simple in a python iter world :). Can we change the string to "Computing non-self-consistent VV10 Energy Correction", would match literature better.

Reply:
Good call. My first associations to earlier string were with correlated methods.